EIPS/eip-5528.md

```diff
@@ -0,0 +1,183 @@
+- The issuer creates an escrow smart contract with detailed escrow information. The information could include issuer token contract address, buyer token contract address,  lock period, exchange rate, the maximum number of buyers, minimum balance of buyers and, etc.
+- The issuer funds issuer tokens to the escrow contract.
+- Buyers fund buyer tokens which are pre-defined in the escrow contract.
+- When the escrow status meets success, the seller can withdraw buyer tokens and buyers can withdraw seller tokens based on exchange rates.
+- Buyers can withdraw(or refund) their funded token if the escrow process is failed or is in the middle of the escrow process.
+
+We have suggested this process be possible in an on-chain network with a payable currency-like token(ex: USDT).
+
+## Motivation
+
+Escrow service holds the money until a particular condition has been met for the seller and buyer.  By `ERC5528` standard, smart contract developers can define a wide range of rules to make the deals more successful.
+
+## Specification
+
+There are 3 contracts for the escrow process: `Buyer Contract`, `Seller Contract`, and `Escrow Contract`.
+ - Buyer Contract: Buyers will pay to an escrow account to exchange with `Seller Token`.
+ - Seller Contract: The seller will pay to the escrow account to exchange with `Buyer Token`.
+ - Escrow Contract: Will be created by the seller. The contract source code allows users(seller and buyers) based on constraint rules. Instead of a simple address mapped balance variable in [EIP-20](./eip-20.md) tokens, the user’s balance should be [Seller Token, Buyer Token].
+
+
+```solidity
+pragma solidity ^0.4.20;
+
+
+interface ERC5528 is ERC20 {
+
+    /// @notice escrow balance of owner
+    /// @dev assigned to the zero address is considered invalid, and this
+    ///   function throws for queries about the zero address.
+    /// @param
+    ///   - _owner: An address for whom to query the balance
+    /// @return amount of current escrow account balance.
+    ///   - in case of an escrow contract, it can be the seller's token or buyer's token for backward compatibility with the ERC20 standard.
+    ///   - in case of seller/buyer contract, same as other ERC20 standards.
+    function balanceOf(address account) public view returns (uint256);
+
+
+    /// @notice escrow balance of owner
+    /// @dev assigned to the zero address is considered invalid, and this
+    ///   function throws for queries about the zero address.
+    /// @param
+    ///   - _owner: An address for whom to query the balance
+    /// @return amount of current escrow account balance. First is buyer token , and seconds is seller token
+    function escrowBalanceOf(address account) public view returns (uint256, uint256);
+
+
+    /// @notice simple query to return a simple description of compliance.
+    /// @dev must be implemented in Escrow-Contract and optional for other contracts.
+    function escrowComplianceDescription() external view returns (string);
```

Comment:
Recommended: Doesn't this introduce a phishing risk?
Recommended: Please make this a URI thing like EIP-721's `tokenURI`